src/Compilers/CSharp/Portable/LanguageVersion.cs

```diff
@@ -121,4 +104,107 @@ internal static ErrorCode GetErrorCode(this LanguageVersion version)
             }
         }
     }
-}
+
+    /// <summary>
+    /// This type is attached to diagnostics for required language version and should only be used
+    /// on such diagnostics, as they are scrapped out by <see cref="CSharpCompilation.GetRequiredLanguageVersion"/>.
+    /// </summary>
+    internal class RequiredLanguageVersion : IMessageSerializable
+    {
+        internal LanguageVersion Version { get; }
+
+        internal RequiredLanguageVersion(LanguageVersion version)
+        {
+            Version = version;
+        }
+
+        public override string ToString()
+        {
+            return Version.ToDisplayString();
+        }
+    }
+
+    public static class LanguageVersionExtensions
+    {
+        /// <summary>
+        /// Displays the version number in the format expected on the command-line (/langver flag).
+        /// For instance, "6", "7", "7.1", "latest".
+        /// </summary>
+        public static string ToDisplayString(this LanguageVersion version)
+        {
+            switch (version)
+            {
+                case LanguageVersion.CSharp1:
+                    return "1";
+                case LanguageVersion.CSharp2:
+                    return "2";
+                case LanguageVersion.CSharp3:
+                    return "3";
+                case LanguageVersion.CSharp4:
+                    return "4";
+                case LanguageVersion.CSharp5:
+                    return "5";
+                case LanguageVersion.CSharp6:
+                    return "6";
+                case LanguageVersion.CSharp7:
+                    return "7";
```

Comment:
Support for 7.1 is happening in the other PR? I think so, just trying to wrap my head around it.